nautobot/extras/filters/__init__.py

```diff
@@ -448,10 +452,44 @@ class NautobotFilterSet(
 
 
 class ContactFilterSet(NameSearchFilterSet, NautobotFilterSet):
+    similar_contacts = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=Location.objects.all(),
+        label="Similar contacts",
+        method="_similar_contacts",
+    )
+
     class Meta:
         model = Contact
         fields = "__all__"
 
+    def generate_query__similar_contacts(self, value):
+        """Helper method used by Contacts and by _similar_to() method."""
+        query_params = Q()
+        if value:
+            location = value[0]
+            contact_name = location.contact_name
+            contact_phone = location.contact_phone
+            contact_email = location.contact_email
+            if contact_name:
+                contact_names = list(Contact.objects.all().values_list("name", flat=True))
+                name_matches = get_close_matches(contact_name, contact_names, cutoff=0.9)
+                query_params |= Q(name__in=name_matches)
+            if contact_phone:
+                contact_phones = list(Contact.objects.all().values_list("phone", flat=True))
+                phone_matches = get_close_matches(contact_phone, contact_phones, cutoff=0.9)
+                query_params |= Q(phone__in=phone_matches)
+            if contact_email:
+                contact_emails = list(Contact.objects.all().values_list("email", flat=True))
```

Comment:
Could/should we maybe do a single `.values_list("name", "phone", "email")` and then split it out into separate lists Python-side so that we don't need to hit the DB three separate times?

I'm also a bit concerned about the performance of this with, say, 1000 existing Contacts in the database?

Reply:
I'll test with 10000 locations, 1000 contacts and 1000 teams in the database and see what the performance is like.

Reply:
Seems pretty snappy to me

![image](https://github.com/nautobot/nautobot/assets/75227981/db2e1754-893f-4ac5-81b3-ee5e3b4d9513)

Reply:
Good enough for me!

Reply:
Here it is running through a first draft of the management command, just running the two filters for each location. It used up about 200MB of ram total between the DB and Nautobot and I think it was bottlenecked by the DB because it was sitting around 80% CPU on the postgres container and 20% on the Nautobot container the entire time.

```
Processed 10007 locations, found 905 locations with similar contacts or teams.
Took 1352.8843789100647 seconds.
```